### PR TITLE
Migrate to new CDB remote config

### PR DESCRIPTION
**Pull request by TurpIF**

### CODE REVIEW

File: CriteoPublisherSdk/Sources/Configuration/CR_RemoteConfigRequest.h [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+@class CR_Config;
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface CR_RemoteConfigRequest : NSObject
+
+@property(copy, nonatomic, readonly) NSString *configUrl;
+
++ (instancetype)createFromConfig:(CR_Config *)config;
```
Comment (vguerci) [inline CriteoPublisherSdk/Sources/Configuration/CR_RemoteConfigRequest.h:30]: Naming conventions: Should be `[CR_RemoteConfigRequest remoteConfigRequestWithConfig:]` but `[CR_RemoteConfigRequest requestWithConfig:]` is ok too

File: CriteoPublisherSdk/Sources/Configuration/CR_RemoteConfigRequest.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+- (NSString *)toGetQueryString;
```
Comment (vguerci) [inline CriteoPublisherSdk/Sources/Configuration/CR_RemoteConfigRequest.h]: Naming: `queryString` is enough. Could also be declared as a readonly property, and potentially initialized early.

File: CriteoPublisherSdk/Sources/Configuration/CR_ConfigManager.m [hunk 3]
```diff
@@ -40,7 +41,8 @@ - (instancetype)initWithApiHandler:(CR_ApiHandler *)apiHandler
 }
 
 - (void)refreshConfig:(CR_Config *)config {
-  [self->apiHandler getConfig:config
+  CR_RemoteConfigRequest *remoteConfigRequest = [CR_RemoteConfigRequest createFromConfig:config];
```
Comment (vguerci) [inline CriteoPublisherSdk/Sources/Configuration/CR_ConfigManager.m:44]: Unless there are other request objects around just naming this var `request` is enough (same goes for the whole commit)

File: CriteoPublisherSdk/Tests/UnitTests/Configuration/CR_ConfigManagerTests.m [hunk 4]
```diff
@@ -106,7 +106,7 @@ - (void)prepareApiHandlerToRespondRemoteConfigJson:(NSString *)jsonResponse {
   NSData *dataResponse = [jsonResponse dataUsingEncoding:NSUTF8StringEncoding];
   NSDictionary *dictionaryResponse = [CR_Config getConfigValuesFromData:dataResponse];
 
-  OCMStub([mockApiHandler getConfig:localConfig
+  OCMStub([mockApiHandler getConfig:OCMOCK_ANY
```
Comment (vguerci) [inline CriteoPublisherSdk/Tests/UnitTests/Configuration/CR_ConfigManagerTests.m:109]: :iseewhatyoudidthere:

File: CriteoPublisherSdk/Sources/Configuration/CR_RemoteConfigRequest.h [hunk 5]
```diff
@@ -29,7 +29,7 @@ NS_ASSUME_NONNULL_BEGIN
 
 + (instancetype)createFromConfig:(CR_Config *)config;
 
-- (NSString *)toGetQueryString;
+- (NSDictionary *)toPostPayload;
```
Comment (vguerci) [inline CriteoPublisherSdk/Sources/Configuration/CR_RemoteConfigRequest.h:32]: Naming: `postBody` or `postData`

File: CriteoPublisherSdk/Sources/Configuration/CR_RemoteConfigRequest.h (hunk 2 above)
Comment (TurpIF) [reply CriteoPublisherSdk/Sources/Configuration/CR_RemoteConfigRequest.h]: done

File: CriteoPublisherSdk/Sources/Configuration/CR_RemoteConfigRequest.h (hunk 1 above)
Comment (TurpIF) [reply CriteoPublisherSdk/Sources/Configuration/CR_RemoteConfigRequest.h]: done

File: CriteoPublisherSdk/Sources/Configuration/CR_ConfigManager.m (hunk 3 above)
Comment (TurpIF) [reply CriteoPublisherSdk/Sources/Configuration/CR_ConfigManager.m]: done

File: CriteoPublisherSdk/Sources/Configuration/CR_RemoteConfigRequest.h (hunk 5 above)
Comment (TurpIF) [reply CriteoPublisherSdk/Sources/Configuration/CR_RemoteConfigRequest.h]: done

**Pull request merged**